app/models/miq_widget_set.rb

```diff
@@ -1,8 +1,27 @@
 class MiqWidgetSet < ApplicationRecord
+  include_concern 'SetData'
+
   acts_as_miq_set
 
+  before_destroy :ensure_can_be_destroyed
   before_destroy :destroy_user_versions
+  before_destroy :delete_from_dashboard_order
+
+  after_save     :add_to_dashboard_order
   before_save    :keep_group_when_saving
+  after_save     :update_members
+
+  validates :group_id, :presence => true, :unless => ->(x) { x.read_only? }
```

Comment:
Think you can do
```suggestion
  validates :group_id, :presence => true, :unless => :read_only?
```

Reply:
good call jason.

Also, thanks Libor for validating group_id and not group - much better for performance in production but can act strange in development with factories